sdk/eventhub/event-hubs/src/connectionContext.ts

```diff
@@ -480,13 +496,21 @@ export function createConnectionContext(
       options = credentialOrOptions;
     }
 
-    // Since connectionstring was passed, create a SharedKeyCredential
-    credential = SharedKeyCredential.fromConnectionString(connectionString);
+    const parsed = parseEventHubConnectionString(connectionString) as Required<
+      | Pick<EventHubConnectionStringProperties, "sharedAccessKey" | "sharedAccessKeyName">
+      | Pick<EventHubConnectionStringProperties, "sharedAccessSignature">
+    >;
+    // Since connectionString was passed, create a TokenProvider.
+    credential = createSasTokenProvider(parsed);
   } else {
     // host, eventHubName, a TokenCredential and/or options were passed to constructor
     const eventHubName = eventHubNameOrOptions;
     let host = hostOrConnectionString;
-    credential = credentialOrOptions;
+    if (isNamedKeyCredential(credentialOrOptions) || isSASCredential(credentialOrOptions)) {
+      credential = new SasTokenProviderImpl(credentialOrOptions);
```

Comment:
Just a thought... If we use `createSasTokenProvider` here as well, then we don't need to export `SasTokenProviderImpl` from core-amqp

Reply:
Great idea!